### PR TITLE
[google_maps_flutter] Allow BitmapDescriptor scaling override

### PR DESCRIPTION
**Pull request by duzenko**
## Description

Patch for BitmapDescriptor.fromAssetImage that respects ImageConfiguration.devicePixelRatio

## Related Issues

https://github.com/flutter/flutter/issues/34386

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [ ] My PR includes unit or integration tests for *all* changed/updated/fixed behaviors (See [Contributor Guide]).
- [x] All existing and new tests are passing.
- [ ] I updated/added relevant documentation (doc comments with `///`).
- [x] The analyzer (`flutter analyze`) does not report any problems on my PR.
- [x] I read and followed the [Flutter Style Guide].
- [x] The title of the PR starts with the name of the plugin surrounded by square brackets, e.g. [shared_preferences]
- [ ] I updated pubspec.yaml with an appropriate new version according to the [pub versioning philosophy].
- [ ] I updated CHANGELOG.md to add a description of the change.
- [x] I signed the [CLA].
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Does your PR require plugin users to manually update their apps to accommodate your change?

- [ ] Yes, this is a breaking change (please indicate a breaking change in CHANGELOG.md and increment major revision).
- [x] No, this is *not* a breaking change.

[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/flutter/plugins/blob/master/CONTRIBUTING.md
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[pub versioning philosophy]: https://www.dartlang.org/tools/pub/versioning
[CLA]: https://cla.developers.google.com/

### CODE REVIEW

File: packages/google_maps_flutter/lib/src/bitmap.dart [hunk 1]
```diff
@@ -59,6 +59,13 @@ class BitmapDescriptor {
     AssetBundle bundle,
     String package,
   }) async {
+    if (configuration.devicePixelRatio != null) {
+      return BitmapDescriptor._(<dynamic>[
+        'fromAssetImage',
+        assetName,
+        configuration.devicePixelRatio,
+      ]);
+    }
```
Comment (iskakaushik) [inline packages/google_maps_flutter/lib/src/bitmap.dart]: line 76 can be `configuration.devicePixelRatio ?? assetBundleImageKey.scale`

**Comment by iskakaushik** [comment]
please also change the pubspec.yml and changelog files.

File: packages/google_maps_flutter/lib/src/bitmap.dart (hunk 1 above)
Comment (duzenko) [reply packages/google_maps_flutter/lib/src/bitmap.dart]: That would invoke unnecessary calls to 
```
    final AssetImage assetImage =
        AssetImage(assetName, package: package, bundle: bundle);
    final AssetBundleImageKey assetBundleImageKey =
        await assetImage.obtainKey(configuration);
```
I don't think we need to waste CPU time and battery doing that when scaling is already provided.

File: packages/google_maps_flutter/CHANGELOG.md [hunk 2]
```diff
@@ -1,3 +1,7 @@
+## 0.5.17+1
```
Comment (iskakaushik) [inline packages/google_maps_flutter/CHANGELOG.md]: Can you change this to `0.5.19` please. Thanks

File: packages/google_maps_flutter/lib/src/bitmap.dart (hunk 1 above)
Comment (iskakaushik) [reply packages/google_maps_flutter/lib/src/bitmap.dart]: Makes sense.

File: packages/google_maps_flutter/CHANGELOG.md (hunk 2 above)
Comment (duzenko) [reply packages/google_maps_flutter/CHANGELOG.md]: Done

**Comment by duzenko** [comment]
I would still like to have a sync and/or contextless version of this function

**Comment by asanoki** [comment]
This CL breaks loading assets of correct scale. It passes the path to 1.0 scaled asset, and the scale ratio to Java, where however the scale is discarded. So even if mipmaps are available, always the 1.0x asset will be loaded. Can we revert?

**Comment by duzenko** [comment]
> 
> 
> This CL breaks loading assets of correct scale. It passes the path to 1.0 scaled asset, and the scale ratio to Java, where however the scale is discarded. So even if mipmaps are available, always the 1.0x asset will be loaded. Can we revert?

1. Can you attach a repro case/pull request for the example app
2. Did you try configuration.devicePixelRatio = null?

**Comment by asanoki** [comment]
> This change should only kick in when devicePixelRatio != null which is exactly the idea behind it so what's broken?

That's true, but in order to load the correct mipmap on Flutter, I can't pass null. Here's what I'm doing:

    final imageConfiguration = createLocalImageConfiguration(context);
    _markerIcon = await BitmapDescriptor.fromAssetImage(
        imageConfiguration, 'assets/images/marker.png');

With this CL, mipmap of scale 1.0 will be loaded, even though the imageConfiguration has scale 3.0. Something is clearly broken, as the incorrect asset is loaded when you pass a scale in ImageConfiguration.

> Did you try configuration.devicePixelRatio = null?

Passing null doesn't help, as then mipmaps won't be used. Just tried. Also, not sure why would I override devicePixelRatio with null, if the one provided by Flutter's createLocalImageConfiguration is correct. It's just after this CL, the value is ignored.

**Comment by duzenko** [comment]
Right
So what you need is a BitmapDescriptor of the image mipmap closest to given pixel ratio
While I need a BitmapDescriptor of the image with given pixel ratio.
This is a breaking change. I will commit a resolve now.

**Pull request merged**